cobigen/cobigen-core-parent/cobigen-core/src/main/java/com/devonfw/cobigen/impl/generator/InputInterpreterImpl.java

```diff
@@ -68,23 +70,79 @@ public Object read(String type, Path path, Charset inputCharset, Object... addit
     public Object read(Path path, Charset inputCharset, Object... additionalArguments) throws InputReaderException {
         Set<String> keySet = PluginRegistry.getTriggerInterpreterKeySet();
         // We first try to find an input reader that is most likely readable
+        Map<String, Boolean> readableCache = new HashMap<>();
+        Object readable = null;
+
+        // Create cache for readable states
         for (String s : keySet) {
-            try {
-                if (isMostLikelyReadable(s, path)) {
-                    LOG.debug("Try reading input {} with inputreader '{}'...", path, s);
+            readable = getValidInputReader(path, inputCharset, readableCache, s, additionalArguments);
+            if (readable != null) {
+                return readable;
+            }
+            readableCache.put(s, isMostLikelyReadable(s, path));
```

Comment:
you called now isMostLikelyReadable three times each iteration :)

Reply:
adjusted.